### DISCUSSION TITLE
API Errors

### DISCUSSION THREAD

**Discussion by Jinsin456** [jbergler/hass-ttlock#98] [Q&A] [answered]
I'm finding I'm getting a lot of sporadic API Errors from the unlock call service

I thought my gateway was too far away from the door but I've checked the logs and HA has tried unlocking the door but there is nothing in the TT Lock logs to say that it's been attempted 

It doesn't seem to happen all the time and I can't detect a pattern. I'm calling the service from Node Red but when it fails I'll call it from my lock icon on the dashboard and it comes up API error / failed etc

Could it be the gateway is too far away? It always seems to work when manually using the app too

Btw I'm using the G2 gateway and a Simpler lock

**Comment by jbergler** [comment]
There's very little I can do to help dig into this unfortunately.

Do you also have the same problems unlocking from the official app with bluetooth disabled (to force the app to use the API + Gateway)?

**Comment by Jinsin456** [comment, marked as answer]
I think this was actually an issue with the distance of the gateway from the door. Where my gateway was, it was in an area of the house where the wifi isn't great and also not the closest to the door.

I've since upgraded to a POE gateway and have it right next to the door, it's been perfect ever since
